src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -67,6 +64,211 @@ case class Task(
 }
 
 object Task {
+  private[this] def newHealthStatus(
+    current: MesosProtos.TaskStatus,
+    update: MesosProtos.TaskStatus): Option[MesosProtos.TaskStatus] = {
+
+    val healthy = update.hasHealthy && (!current.hasHealthy || current.getHealthy != update.getHealthy)
+    val changed = healthy || current.getState != update.getState
+    if (changed) {
+      Some(update)
+    }
+    else {
+      None
+    }
+  }
+
+  /**
+    * A LaunchedEphemeral task is a stateless task that does not consume reserved resources or persistent volumes.
+    */
+  case class LaunchedEphemeral(
+      taskId: Task.Id,
+      agentInfo: AgentInfo,
+      appVersion: Timestamp,
+      status: Status,
+      networking: Networking) extends Task {
+
+    private[this] val log = LoggerFactory.getLogger(getClass)
```

Comment:
Put the logger into the companion object. No need to resolve the logger for every task instance.